repos/system_upgrade/el7toel8/libraries/repository_scanning.py

```diff
@@ -0,0 +1,88 @@
+import json
```

Comment:
I'd name the file `repofileutils.py` (you can see various Xutils.py libraries there already) or maybe even better just `repofile.py`, similarly to `rhsm.py`.